modules/json_content/src/Plugin/Derivative/EntityLocalTasks.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Drupal\json_content\Plugin\Derivative;
+
+use Drupal\Component\Plugin\Derivative\DeriverBase;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Plugin\Discovery\ContainerDeriverInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines dynamic local tasks.
+ */
+class EntityLocalTasks extends DeriverBase implements ContainerDeriverInterface {
+
+  /**
+   * The route match.
+   *
+   * @var RouteMatchInterface
+   */
+  protected $routeMatch;
+
+  public function __construct(RouteMatchInterface $route_match) {
+    $this->routeMatch = $route_match;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, $base_plugin_id) {
+    return new static(
+      $container->get('current_route_match')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDerivativeDefinitions($base_plugin_definition) {
+    $raw_parameters = $this->routeMatch->getRawParameters();
+    $entity_type = reset($raw_parameters->keys());
```

Comment:
Seems like there must be a better way to get the entity object from the route. @phenaproxima ?